server/util/claims/claims.go

```diff
@@ -43,10 +43,15 @@ var (
 
 type Claims struct {
 	jwt.StandardClaims
-	APIKeyID      string `json:"api_key_id,omitempty"`
-	UserID        string `json:"user_id"`
-	GroupID       string `json:"group_id"`
-	Impersonating bool   `json:"impersonating"`
+	APIKeyID string `json:"api_key_id,omitempty"`
+	UserID   string `json:"user_id"`
+	GroupID  string `json:"group_id"`
+	// APIKeyOwnerGroupID identifies the group that owns the API key used
+	// for authentication. Will be empty if authentication was not performed
+	// using an API key.
+	// The GroupID field above should be used in most cases!
+	APIKeyOwnerGroupID string `json:"api_key_owner_group_id,omitempty"`
```

Comment:
could we call this ParentGroupID?

Reply:
If auth is performed using a key in the child group, this field will have the group ID of the child group so it might be misleading to call it ParentGroupID.